src/domain/dataset-update-flow/src/entities/flow/flow_start_condition.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::Duration;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FlowStartCondition {
+    Throttling(FlowStartConditionThrottling),
+    Batching(FlowStartConditionBatching),
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct FlowStartConditionThrottling {
+    pub interval: Duration,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct FlowStartConditionBatching {
+    pub available_records: usize,
```

Comment:
This isn't used yet. I removed, but we will have to have some source of truth in this regard.
Following the example with the IoT device, we still have to create an every-second secondary trigger event and must show each in the History view.

Currently this trigger is an empty struct:

```
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct FlowTriggerPush {
    // TODO: source (HTTP, MQTT, CMD, ...)
}
```

similarly, we don't store yet the number of records added for the flow outcome, and can't use it yet for the derived dataset throttling:

```

#[derive(Debug, Clone, PartialEq, Eq)]
pub struct FlowTriggerInputDatasetFlow {
    pub input_dataset_id: DatasetID,
    pub input_flow_type: DatasetFlowType,
    pub input_flow_id: FlowID,
}
```

Perhaps both these structures should have information about the offset change. Sum of these numbers might be the criteria we are looking for.

Reply:
Hm, I was imaging it differently:
- Kafka controller will know about batching configuration
- When Kafka queue becomes non-empty - it will trigger the flow (to show users that batch is pending)
- While more records are flowing - it will NOT touch the flow system, until the configured batch size is reached to finally trigger the flow

This way secondary trigger events will appear only for other types of trigger, e.g. "manual".

While the flow is already in the state of waiting for a batch - I don't think showing more data arriving as secondary triggers is useful, and can result in too many updates.

Reply:
Assuming this suggestion, without the visibility of Kafka's queue state, let's say, showing something like  `accumulated 17,245/25,000 records`, the user will not understand what is the flow waiting for.